src/jit/flowgraph.cpp

```diff
@@ -19273,16 +19266,7 @@ unsigned Compiler::fgGetCodeEstimate(BasicBlock* block)
     for (Statement* stmt : StatementList(block->FirstNonPhiDef()))
     {
         unsigned char cost = stmt->GetCostSz();
-        if (cost < MAX_COST)
-        {
-            costSz += cost;
-        }
-        else
-        {
-            // We could walk the tree to find out the real GetCostSz(),
-            // but just using MAX_COST for this trees code size works OK
-            costSz += cost;
-        }
+        costSz += cost;
```

Comment:
The original code here seems suspect. If we're only comparing `cost` to `MAX_COST` that doesn't ensure that the sum won't overflow, in which case we're not actually using `MAX_COST`.

Reply:
GetCosts can't return a value bigger than `MAX_COST`, we check that when we set this value:
https://github.com/dotnet/coreclr/blob/eae780cfa390ba1ea26f575a7082b961ea473864/src/jit/gentree.h#L461-L469